File: 프로그래머스/Lv.2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    const answer = []
    let left = 0
    let right = 0
    let sum = sequence[0]             // sum 3 + 4 = 7
                                      //                    l  r
    while (right < sequence.length) { // sequence = [ 1, 2, 3, 4, 5]
        if (sum < k) {
            right++
            sum += sequence[right]
        } else if (sum > k) {
            sum -= sequence[left]
            left++
        } else {
            answer.push([left, right])
            right++
            sum += sequence[right]
        }
    }
    
    const sorted = answer.sort((a, b) => {
       const lengthDiff = (a[1] - a[0]) - (b[1] - b[0])
       if (lengthDiff !== 0) return lengthDiff
        else a[0] - b[0]
    })
    
    return sorted[0]
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    
    curSum = 0
    left = 0
    right = 0
    minLength = len(sequence)
    
    for right in range(len(sequence)): # right이 끝까지 갈 때까지
        curSum += sequence[right]
        
        while k < curSum:
            curSum -= sequence[left]
            left += 1
            
        # 현재 부분 수열의 합이 k와 같을 때
        if k == curSum:
            if right - left < minLength: # 이전 발견한 수열 보다 길이가 짧은 경우에만 갱신
                minLength = right - left
                answer = [left, right]
    return answer